### PR TITLE
Removes utilization check from rebalancer

### PR DESCRIPTION
**Pull request by dposada**
## Changes proposed in this PR

- removing the `min-utilization-threshold` config option
- removing the `mesos-master-hosts` config option
- removing `cook.mesos/get-mesos-utilization`
- in the rebalancer, removing the utilization check (i.e. run the rebalancer regardless of cluster utilization)

## Why are we making these changes?

The utilization check was useful before the rebalancer took "spare" resources into account when determining if tasks need to be preempted. Now that these "spare" resources are accounted for (via incubating offers), it doesn't make much sense to only rebalance when cluster utilization is below a certain threshold.

### CODE REVIEW

File: integration/tests/cook/util.py [hunk 1]
```diff
@@ -318,13 +318,10 @@ def retrieve_mesos_url(varname='MESOS_PORT', value='5050'):
     mesos_url = os.getenv('COOK_MESOS_LEADER_URL')
     if mesos_url is None:
         mesos_port = os.getenv(varname, value)
-        cook_url = retrieve_cook_url()
-        _wait_for_cook(cook_url)
-        mesos_master_hosts = settings(cook_url).get('mesos-master-hosts', ['localhost'])
-        resp = session.get('http://%s:%s/redirect' % (mesos_master_hosts[0], mesos_port), allow_redirects=False)
+        resp = session.get(f'http://localhost:{mesos_port}/redirect', allow_redirects=False)
```
Comment (wyegelwel) [inline integration/tests/cook/util.py]: Why is it ok to assume localhost here? I'm imagining a case where you run the integration tests on a production cluster and then the tests that check the number of hosts. What am I missing?

File: scheduler/docs/configuration.adoc [hunk 2]
```diff
@@ -111,10 +111,6 @@ We'll look at the configurable options in turn:
   This option sets the Mesos master connection string.
   For example, if you are running Mesos with a Zookeeper node on the local machine (a common development setup), you'd use the connection string `zk://localhost:2181/mesos`.
 
-`:master-hosts`::
```
Comment (wyegelwel) [inline scheduler/docs/configuration.adoc]: Do you think it is worthwhile to have a deprecated or removed section in our config doc so users who upgrade aren't confused why the setting they used to use is no longer documented. Perhaps as an alternative, having the init in the scheduler print that the config is no longer used? 

I don't think it is a big deal so I'm kind of on the fence about it.

File: integration/tests/cook/util.py (hunk 1 above)
Comment (dposada) [reply integration/tests/cook/util.py]: You are correct, this is only a decent assumption when running locally. For other environments, we will have to set the `COOK_MESOS_LEADER_URL` environment variable (see 3 lines above).

File: scheduler/docs/configuration.adoc (hunk 2 above)
Comment (dposada) [reply scheduler/docs/configuration.adoc]: I like the idea of printing that it's no longer used. I will add that.

File: integration/tests/cook/util.py (hunk 1 above)
Comment (wyegelwel) [reply integration/tests/cook/util.py]: ah ha! That is what I was missing. Thanks, I knew it was something stupid on my part.

File: scheduler/docs/configuration.adoc (hunk 2 above)
Comment (dposada) [reply scheduler/docs/configuration.adoc]: Done

File: scheduler/src/cook/config.clj [hunk 3]
```diff
@@ -267,17 +267,10 @@
                             (when scheduler
                               (or (:good-enough-fitness scheduler) 0.8)))
      :mesos-master (fnk [[:config {mesos nil}]]
+                     (when (:master-hosts mesos)
+                       (log/warn "The :master-hosts configuration field is no longer used"))
```
Comment (shamsimam) [inline scheduler/src/cook/config.clj]: 👍

**Pull request merged**